lib/inputs/rpm/static.ts

```diff
@@ -0,0 +1,88 @@
+import { unlink, writeFile } from "fs";
+import { EOL, tmpdir } from "os";
+import { resolve as resolvePath } from "path";
+import { getContent } from "../../extractor";
+import { ExtractAction, ExtractedLayers } from "../../extractor/types";
+import { streamToBuffer } from "../../stream-utils";
+import { CmdOutput, execute } from "../../sub-process";
+import { RpmDbPackage } from "./types";
+
+export const getRpmDbFileContentAction: ExtractAction = {
+  actionName: "rpm-db",
+  fileNamePattern: "/var/lib/rpm/Packages",
+  callback: streamToBuffer,
+};
+
+export async function getRpmDbFileContent(
+  extractedLayers: ExtractedLayers,
+): Promise<string> {
+  const apkDb = getContent(extractedLayers, getRpmDbFileContentAction);
+  if (!apkDb) {
+    return "";
+  }
+
+  const filePath = generateTempFileName();
+  await writeToFile(filePath, apkDb as Buffer);
+
+  try {
+    const cmdOutput = await execute("rpmdb", ["-f", filePath]).catch(
+      handleError,
+    );
+    return formatRpmDbOutput(cmdOutput);
+  } finally {
+    await removeFile(filePath);
+  }
+}
+
+function handleError(error): CmdOutput | never {
+  const stderr = error.stderr;
+  if (typeof stderr === "string" && stderr.indexOf("not found") >= 0) {
+    return { stdout: "", stderr: "" };
+  }
+  throw error;
+}
+
+function generateTempFileName(): string {
+  const tmpPath = tmpdir();
+  const randomFileName = Math.random().toString();
+
+  return resolvePath(tmpPath, randomFileName);
+}
+
+function writeToFile(filePath: string, apkDb: Buffer): Promise<string> {
+  return new Promise((resolve, reject) => {
+    writeFile(filePath, apkDb, { encoding: "binary" }, (err) => {
+      if (!err) {
+        resolve(filePath);
+      } else {
+        reject(err);
+      }
+    });
+  });
+}
+
+function removeFile(filePath: string): Promise<void> {
+  return new Promise((resolve) => {
+    unlink(filePath, () => {
+      resolve();
+    });
+  });
+}
+
+/**
+ * Exported for testing
+ */
+export function formatRpmDbOutput(output: CmdOutput): string {
+  if (!output.stdout) {
```

Comment:
shouldn't we throw here?

Reply:
No because the caller of "get RPM DB content" decides whether to throw: by returning empty it can examine the other package manager outputs. Only when all are empty (no package manager detected) then the caller throws